### DISCUSSION TITLE
Change default for: Minimum acceptable expiration time for BOLT11 for refunds

### DISCUSSION THREAD

**Discussion by mp21-org** [btcpayserver/btcpayserver#3686] [Ideas / Feature Requests]
Following up with the discussion on [mattermost](https://chat.btcpayserver.org/btcpayserver/pl/x4a6kkyb8f88iy6ir9mfn9nfgh).

We should consider a different default for this setting. Most customer lightning wallets can only provide 24h valid invoices.
Thanks for considering it.

**Comment by Kukks** [comment]
It's already possible to change the accepted minimum export date though?

**Reply by pavlenex** [reply]
The default is bad https://twitter.com/BtcPins/status/1519752108785864705?s=20&t=rpuPawFQxiwHPMEBCLwTIQ 

Good default mostly would mean it works for most people, the 30 day default doesn't work with any wallet.

**Reply by mp21-org** [reply]
As mentioned in the mattermost discussion it can be changed in the store settings, yes - but the default doesn't make any sense.

**Reply by mp21-org** [reply]
I think there is also a bug in this somewhere, at least with 1.5.1.0 I have a strange behavoiur:

I have 2 stores, one with the default 30 days (1st store) and one where I changed it to 1 day:
![image](https://user-images.githubusercontent.com/1374810/166136872-4252beb2-c1a3-47df-81e7-2b80eb597b6d.png)

When I create a pull-payment on the second store with the 1 day value, I can't claim it and get the error that it needs to be at least a 30 day valid invoice:

`The BOLT11 invoice must have an expiry date of at least 30 days from submission (Provided was only 13).`

If someone can confirm/reproduce this, I would open a bug report for it.

**Reply by bolatovumar** [reply]
@petzsch looks like the store setting for minimum acceptable BOLT11 expiration time doesn't have anything to do with pull payments at all, it only applies to refunds of invoices.

For pull payments there is an option you can configure on each pull payment separately and it defaults to 30 days. I was very confused about this as well as it's out of the way and it's not obvious that this option even exists.

![Capture](https://user-images.githubusercontent.com/1934678/170176850-4f85e589-42ca-4bda-872b-8f6cea7b7b61.PNG)

**Reply by bolatovumar** [reply]
I made a draft PR to update the default from 30 days to 1 day but I'm not sure if that's enough. Looks like the default as specified in BOLT11 is 3600 seconds i.e. 1 hour. I wonder if we should change it to be the same.

![Capture](https://user-images.githubusercontent.com/1934678/170178500-7ae1cfa0-0b0e-4ba0-9d06-c82d4744cd9c.PNG)

https://github.com/lightning/bolts/blob/master/11-payment-encoding.md

**Reply by bolatovumar** [reply]
Draft PR here: https://github.com/btcpayserver/btcpayserver/pull/3776
